pkg/backends/calico/routes.go

```diff
@@ -15,30 +15,240 @@ package calico
 
 import (
 	"os"
-	"strings"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/kelseyhightower/confd/pkg/resource/template"
 )
 
+// routeGenerator defines the data fields
+// necessary for monitoring the services/endpoints resources for
+// valid service ips to advertise
 type routeGenerator struct {
-	client *client
+	sync.Mutex
+	client                  *client
+	nodeName                string
+	svcInformer, epInformer cache.Controller
+	svcIndexer, epIndexer   cache.Indexer
+	svcRouteMap             map[string]string // maps service name to ip
```

Comment:
This will need to be a `map[string][]string` since a single service can provide multiple IPs that need to be advertised.